src/naive_bayes/categorical.rs

```diff
@@ -6,50 +6,51 @@
 //! Example:
 //!
 //! ```
-//! use smartcore::linalg::naive::dense_matrix::*;
+//! use smartcore::linalg::dense::matrix::DenseMatrix;
 //! use smartcore::naive_bayes::categorical::CategoricalNB;
 //!
 //! let x = DenseMatrix::from_2d_array(&[
-//!              &[3., 4., 0., 1.],
-//!              &[3., 0., 0., 1.],
-//!              &[4., 4., 1., 2.],
-//!              &[4., 2., 4., 3.],
-//!              &[4., 2., 4., 2.],
-//!              &[4., 1., 1., 0.],
-//!              &[1., 1., 1., 1.],
-//!              &[0., 4., 1., 0.],
-//!              &[0., 3., 2., 1.],
-//!              &[0., 3., 1., 1.],
-//!              &[3., 4., 0., 1.],
-//!              &[3., 4., 2., 4.],
-//!              &[0., 3., 1., 2.],
-//!              &[0., 4., 1., 2.],
+//!              &[3, 4, 0, 1],
+//!              &[3, 0, 0, 1],
+//!              &[4, 4, 1, 2],
+//!              &[4, 2, 4, 3],
+//!              &[4, 2, 4, 2],
+//!              &[4, 1, 1, 0],
+//!              &[1, 1, 1, 1],
+//!              &[0, 4, 1, 0],
+//!              &[0, 3, 2, 1],
+//!              &[0, 3, 1, 1],
+//!              &[3, 4, 0, 1],
+//!              &[3, 4, 2, 4],
+//!              &[0, 3, 1, 2],
+//!              &[0, 4, 1, 2],
 //!          ]);
-//! let y = vec![0., 0., 1., 1., 1., 0., 1., 0., 1., 1., 1., 1., 1., 0.];
+//! let y: Vec<u32> = vec![0, 0, 1, 1, 1, 0, 1, 0, 1, 1, 1, 1, 1, 0];
```

Comment:
It is nice to be able to use normal integers for this

Reply:
The problem is that you make a call to `.to_usize()` in `fn log_likelihood` and `fn fit`, and this function will fail if used on negative integer values. Why would you like to support integers?

Reply:
I am sorry, I was trying to say that it is good to be able to use other types different than floats (the constraint that we currently have in the main branch)

Reply:
I see! I thought you want to make sure we can run CategoricalNB on integer target variables. Yes, that's the plan! Floats are great, but we do need to support other types as well. That's one reason why I decided to try new layer on Naive Bayes first, since NB methods do not apply to floating-point data most of the time.